# Graded assignment 1

"Wout van Iperen #u974489#"


# Packages

We need the package `tidyverse`

In [5]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [6]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3         V4        V5        V6        V7       
1 0.6946172 0.4849023 0.02191465 0.2755784 0.5799686 0.2831372 0.3855690
2 0.2709048 0.3327968 0.60389128 0.4386254 0.2472935 0.3053236 0.4866175
  V8        V9        V10       V11       V12       V13       V14      
1 0.6219855 0.3029043 0.2670511 0.3965684 0.9677481 0.5772012 0.4494684
2 0.9637801 0.9213692 0.1259389 0.5460409 0.4661348 0.7067663 0.2442279
  V15       V16       V17       V18        V19       V20       
1 0.2518073 0.5234542 0.2523934 0.45088152 0.8869653 0.09279698
2 0.3009266 0.3421870 0.6084181 0.03357418 0.3544139 0.86944399

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [8]:
df2 <- df1 %>%
  gather(Key, Value, V1:V20)

head(df2)

Key Value     
1 V1  0.69461718
2 V1  0.27090475
3 V2  0.48490226
4 V2  0.33279684
5 V3  0.02191465
6 V3  0.60389128

## Assignment 2

We will generate some data:

In [9]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [10]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [11]:
df5 <- full_join(df3, df4)
df5

Joining, by = "id"


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook